src/content/reference/react-dom/client/createRoot.md

```diff
@@ -45,7 +45,9 @@ An app fully built with React will usually only have one `createRoot` call for i
 
 * **optional** `options`: An object with options for this React root.
 
-  * **optional** `onRecoverableError`: Callback called when React automatically recovers from errors.
+  * <CanaryBadge title="This feature is only available in the Canary channel" /> **optional** `onCaughtError`: Callback called when React catches an error in an error boundary. Called with the `error` and an `errorInfo` object containing the `componentStack`.
+  * <CanaryBadge title="This feature is only available in the Canary channel" /> **optional** `onUncaughtError`: Callback called when an uncaught error is thrown. Called with the `error` and an `errorInfo` object containing the `componentStack`.
```

Comment:
Incoming "well ackshually": "Throwing an uncaught error" is technically not really a thing. You don't know yet if an error is caught or not when you throw it.

```suggestion
  * <CanaryBadge title="This feature is only available in the Canary channel" /> **optional** `onUncaughtError`: Callback called when a thrown error is not caught. Called with the `error` and an `errorInfo` object containing the `componentStack`.
```

Reply:
well ackshually, react catches the error in the component, and rethrows the error if it's uncaught by an error bounday, so _technically_ react re-throws the uncaught error. But from the user's perspective you're right, it should be "when a thrown error is not caught" 😜

Reply:
Fixed. I also update the descriptions to say:
- uncaught: Called with the `error` that was thrown
- caught: Called with the `error` caught by the Error Boundary
- recoverable: Called with the `error` React throws